input/elasticapm/processor.go

```diff
@@ -363,7 +364,12 @@ func (p *Processor) handleStream(
 
 // processBatch processes the batch and returns it to the pool after it's been processed.
 func (p *Processor) processBatch(ctx context.Context, processor modelpb.BatchProcessor, batch *modelpb.Batch) error {
-	defer p.batchPool.Put(batch)
+	defer func() {
+		for i := range *batch {
+			(*batch)[i] = nil
+		}
+		batchPool.Put(batch)
+	}()
```

Comment:
```suggestion
	defer batchPool.Put((*batch)[:0])
```

Would this work?

Reply:
No, it wouldn't solve the issue because the underlying array would still be kept fully in memory, keeping references to unused `APMEvent`s

From https://go.dev/blog/slices-intro:

> As mentioned earlier, re-slicing a slice doesn’t make a copy of the underlying array. The full array will be kept in memory until it is no longer referenced.

But since we're reusing the `modelpb.Batch` the unerlying array is always reused and kept alive.